offer-manager-domain/src/main/groovy/life/qbic/business/products/Converter.groovy

```diff
@@ -102,16 +163,46 @@ class Converter {
 
     static life.qbic.business.products.Product convertDTOtoProduct(Product product){
         ProductCategory category = getCategory(product)
-        return new life.qbic.business.products.Product.Builder(category,
-                                                                product.productName,
-                                                                product.description,
-                                                                product.unitPrice,
-                                                                product.unit)
-                                                                .build()
+        if (product.internalUnitPrice == 0 && product.externalUnitPrice == 0 && product.unitPrice != 0) { // we used an old constructor
```

Comment:
you could facilitate this statement by introducing a method called instead of that statement.
e.g hasBasePrice() or so

Reply:
I do not think this logic belongs into the product but is the responsibility of the method here.

Reply:
but you use the same logical statement again later on. And this is about the readability of the code and not about responsibilty.

Reply:
on the other hand good point :D it makes it way more readable